src/java.base/share/classes/java/lang/template/TemplateRuntime.java

```diff
@@ -0,0 +1,392 @@
+     */
+    static String interpolate(List<String> fragments, List<Object> values) {
+        Objects.requireNonNull(fragments, "fragments must not be null");
+        Objects.requireNonNull(values, "values must not be null");
+        int fragmentsSize = fragments.size();
+        int valuesSize = values.size();
+        if (fragmentsSize != valuesSize + 1) {
+            throw new RuntimeException("fragments must have one more element than values");
+        }
+        if (fragmentsSize == 1) {
+            String fragment = Objects.requireNonNull(fragments.get(0), "fragments must not have null elements");
+            return fragments.get(0);
+        }
+        int size = fragmentsSize + valuesSize;
+        String[] strings = new String[size];
+        Iterator<String> fragmentsIter = fragments.iterator();
+        int i = 0;
+        for (Object value : values) {
+            strings[i++] = Objects.requireNonNull(fragmentsIter.next(), "fragments must not have null elements");
+            strings[i++] = String.valueOf(value);
+        }
+        strings[i++] = Objects.requireNonNull(fragmentsIter.next(), "fragments must not have null elements");
+        return JLA.join("", "", "", strings, size);
+    }
+
+    /**
+     * Combine one or more {@link StringTemplate StringTemplates} to produce a combined {@link StringTemplate}.
+     * {@snippet :
+     * StringTemplate st = StringTemplate.combine("\{a}", "\{b}", "\{c}");
+     * assert st.interpolate().equals("\{a}\{b}\{c}");
+     * }
+     *
+     * @param sts  one or more {@link StringTemplate}
+     *
+     * @return combined {@link StringTemplate}
+     *
+     * @throws NullPointerException if sts is null or if any element of sts is null
+     */
+    static StringTemplate combine(StringTemplate... sts) {
+        Objects.requireNonNull(sts, "sts must not be null");
+        if (sts.length == 0) {
+            return StringTemplate.of("");
+        } else if (sts.length == 1) {
+            return Objects.requireNonNull(sts[0], "string templates should not be null");
+        }
+        int size = 0;
+        for (StringTemplate st : sts) {
+            Objects.requireNonNull(st, "string templates should not be null");
+            size += st.values().size();
+        }
+        String[] fragments = new String[size + 1];
+        Object[] values = new Object[size];
+        int i = 0, j = 0;
+        fragments[0] = "";
+        for (StringTemplate st : sts) {
+            Iterator<String> fragmentIter = st.fragments().iterator();
+            fragments[i++] += fragmentIter.next();
+            while (fragmentIter.hasNext()) {
+                fragments[i++] = fragmentIter.next();
+            }
+            i--;
+            for (Object value : st.values()) {
+                values[j++] = value;
+            }
+        }
+        return new SimpleStringTemplate(java.lang.template.TemplateRuntime.toList(fragments), java.lang.template.TemplateRuntime.toList(values));
```

Comment:
It seems that IntelliJ was lot when auto-completing or doing a refactoring given that you are alreay in the class TemplateRuntime.